src/main/java/org/apache/tinkerpop/gremlin/tinkergraph/structure/OverflowDbNode.java

```diff
@@ -567,13 +564,17 @@ private int storeAdjacentNode(Direction direction, String edgeLabel, VertexRef<O
    * grow the adjacentVerticesWithProperties array
    *
    * preallocates more space than immediately necessary, so we don't need to grow the array every time
-   * (tradeoff between performance and memory)
+   * (tradeoff between performance and memory).
+   * grows with the square root of the double of the current capacity.
    */
   private Object[] growAdjacentVerticesWithProperties(int offsetPos,
                                                       int strideSize,
                                                       int insertAt,
                                                       int currentLength) {
-    int additionalEntriesCount = (currentLength + strideSize) * growthEmptyFactor;
+    int currentCapacity = currentLength / strideSize;
+    double additionalCapacity = Math.sqrt(currentCapacity * 2) + 1;
```

Comment:
Why are we multiplying by 2 before taking the square root?

Reply:
true, done